### ISSUE TITLE
Cannot start the web application due to aiohttp version compatibility

### ISSUE THREAD

**Issue by pmarcis** [wookayin/gpustat-web#7]
Hi!

I am trying to run the gpustat-web app using python 3.7.3 with gpustat v. 0.6.0, however, after building, installing, and running I get:

```
Traceback (most recent call last):
  File "~/anaconda3/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "~/anaconda3/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "~/gpustat-web/gpustat_web/__main__.py", line 4, in <module>
    main()
  File "~/gpustat-web/gpustat_web/app.py", line 229, in main
    web.run_app(app, host='0.0.0.0', port=args.port)
  File "~/anaconda3/lib/python3.7/site-packages/aiohttp-4.0.0a1-py3.7-linux-x86_64.egg/aiohttp/web.py", line 433, in run_app
    reuse_port=reuse_port))
  File "~/anaconda3/lib/python3.7/asyncio/base_events.py", line 584, in run_until_complete
    return future.result()
  File "~/anaconda3/lib/python3.7/site-packages/aiohttp-4.0.0a1-py3.7-linux-x86_64.egg/aiohttp/web.py", line 305, in _run_app
    await runner.setup()
  File "~/anaconda3/lib/python3.7/site-packages/aiohttp-4.0.0a1-py3.7-linux-x86_64.egg/aiohttp/web_runner.py", line 239, in setup
    self._server = await self._make_server()
  File "~/anaconda3/lib/python3.7/site-packages/aiohttp-4.0.0a1-py3.7-linux-x86_64.egg/aiohttp/web_runner.py", line 352, in _make_server
    await self._app.startup()
  File "~/anaconda3/lib/python3.7/site-packages/aiohttp-4.0.0a1-py3.7-linux-x86_64.egg/aiohttp/web_app.py", line 289, in startup
    await self.on_startup.send(self)
  File "~/anaconda3/lib/python3.7/site-packages/aiohttp-4.0.0a1-py3.7-linux-x86_64.egg/aiohttp/signals.py", line 34, in send
    await receiver(*args, **kwargs)  # type: ignore
  File "~/gpustat-web/gpustat_web/app.py", line 186, in start_background_tasks
    app._tasks = app.loop.create_task(spawn_clients(hosts, exec_cmd, verbose=verbose))
AttributeError: 'Application' object has no attribute 'loop'
```

I did the following:
```
python setup.py build
python setup.py install
python -m gpustat_web --port 48109
```

Any ideas of what I am missing?

**Comment by wookayin** [comment]
I guess aiohttp 4.x have broken the compatibility. Could you try using `aiohttp>=3.0.0,<4.0`?

**Comment by wookayin** [comment]
`app.loop` has been [deprecated since 3.5](https://docs.aiohttp.org/en/latest/changes.html#deprecations-and-removals) and removed from 4.x, so I should fix this.

**Comment by pmarcis** [comment]
Thanks, I got a bit further. Now the SSH connection seems to be failing.
```
Hosts : ['localhost']
Cmd   : gpustat --color --gpuname-width 25
[localhost] type object 'Poly1305' has no attribute 'generate_tag'
Traceback (most recent call last):
  File "~/gpustat-web/gpustat_web/app.py", line 76, in run_client
    await _loop_body()
  File "~/gpustat-web/gpustat_web/app.py", line 51, in _loop_body
    async with asyncssh.connect(host) as conn:
  File "~/anaconda3/lib/python3.7/site-packages/asyncssh-2.1.0-py3.7.egg/asyncssh/misc.py", line 164, in __aenter__
    self._result = await self._coro
  File "~/anaconda3/lib/python3.7/site-packages/asyncssh-2.1.0-py3.7.egg/asyncssh/connection.py", line 5644, in connect
    conn_factory, 'Opening SSH connection to')
  File "~/anaconda3/lib/python3.7/site-packages/asyncssh-2.1.0-py3.7.egg/asyncssh/connection.py", line 170, in _connect
    await conn.wait_established()
  File "~/anaconda3/lib/python3.7/site-packages/asyncssh-2.1.0-py3.7.egg/asyncssh/connection.py", line 1882, in wait_established
    await self._waiter
  File "~/anaconda3/lib/python3.7/site-packages/asyncssh-2.1.0-py3.7.egg/asyncssh/connection.py", line 740, in data_received
    while self._inpbuf and self._recv_handler():
  File "~/anaconda3/lib/python3.7/site-packages/asyncssh-2.1.0-py3.7.egg/asyncssh/connection.py", line 959, in _recv_packet
    processed = handler.process_packet(pkttype, seq, packet)
  File "~/anaconda3/lib/python3.7/site-packages/asyncssh-2.1.0-py3.7.egg/asyncssh/packet.py", line 215, in process_packet
    self._packet_handlers[pkttype](self, pkttype, pktid, packet)
  File "~/anaconda3/lib/python3.7/site-packages/asyncssh-2.1.0-py3.7.egg/asyncssh/kex_dh.py", line 253, in _process_reply
    self._verify_reply(host_key, host_key_data, sig)
  File "~/anaconda3/lib/python3.7/site-packages/asyncssh-2.1.0-py3.7.egg/asyncssh/kex_dh.py", line 227, in _verify_reply
    self._conn.send_newkeys(k, h)
  File "~/anaconda3/lib/python3.7/site-packages/asyncssh-2.1.0-py3.7.egg/asyncssh/connection.py", line 1217, in send_newkeys
    self._send_deferred_packets()
  File "~/anaconda3/lib/python3.7/site-packages/asyncssh-2.1.0-py3.7.egg/asyncssh/connection.py", line 1041, in _send_deferred_packets
    self.send_packet(pkttype, *args)
  File "~/anaconda3/lib/python3.7/site-packages/asyncssh-2.1.0-py3.7.egg/asyncssh/connection.py", line 996, in send_packet
    self.send_packet(MSG_IGNORE, String(b''))
  File "~/anaconda3/lib/python3.7/site-packages/asyncssh-2.1.0-py3.7.egg/asyncssh/connection.py", line 1018, in send_packet
    packet, mac = self._send_encryption.encrypt_packet(seq, hdr, packet)
  File "~/anaconda3/lib/python3.7/site-packages/asyncssh-2.1.0-py3.7.egg/asyncssh/encryption.py", line 189, in encrypt_packet
    return self._cipher.encrypt_and_sign(header, packet, UInt64(seq))
  File "~/anaconda3/lib/python3.7/site-packages/asyncssh-2.1.0-py3.7.egg/asyncssh/crypto/chacha.py", line 138, in encrypt_and_sign
    tag = poly1305(self._key, header + data, nonce)
  File "~/anaconda3/lib/python3.7/site-packages/asyncssh-2.1.0-py3.7.egg/asyncssh/crypto/chacha.py", line 54, in poly1305
    return Poly1305.generate_tag(poly1305_key(key, nonce), data)
AttributeError: type object 'Poly1305' has no attribute 'generate_tag'
```
I tried also adding information about the user name and also passing the exact SSH key (looking at https://asyncssh.readthedocs.io/en/latest/), but with no luck.

Any guess what I might now be missing?

The following versions of libraries were installed:
```
aiohttp                            3.5.0
aiohttp-debugtoolbar               0.6.0
aiohttp-devtools                   0.13.1
aiohttp-jinja2                     1.2.0
asyncssh                           2.1.0
```
Is my asyncssh version too large?

**Comment by wookayin** [comment]
Apparently `Poly1305.generate_tag` was added in pyca/cryptography 2.8+. What is the version of `cryptography` installed on your system? Try upgrading it. If you installed dependency correctly, it should have been automatically taken care of.

**Comment by pmarcis** [comment]
Thanks! Changing to 2.8 fixed that issue. As I use Anaconda (a local python environment), I had to also change the execution command of `gpustats` so that it would point to the exact python version and the exact gpustats path. After changing that, everything works like a charm!

**Comment by wookayin** [comment]
Thanks, great to hear. I would like to have this issue opened until I fully fix the compatibility issue and our aiohttp API usage.

**Comment by wookayin** [comment]
Fixed #7 via fd501df.
